### ISSUE TITLE
Webhooks?

### ISSUE THREAD

**Issue by amaurymartiny** [anonaddy/anonaddy#42]
First, I'd like to start by saying that this product is great, thanks!

I have a request for my use case: would it be possible to add **webhooks**?

In the settings page, the user could add an optional endpoint `https://api.user.personal.website.com`, and, if set, all emails (incl subject, body, attachments) would be sent to this webhook upon receival.

I would like to set up my own personal endpoint, and do some basic parsing operations on the raw emails I receive from others.

If webhooks are set up, this tool could be easily integrated e.g. via Zapier to a wide range of other services. I know it might go against the privacy vision of the app, but again, the webhook will be optional for users.

**Comment by willbrowningme** [comment]
That is a great idea. I do currently have this on my to do list and might try to implement it into the new rules system as one of the actions that can be carried out.

How do you think failed webhook deliveries should be handled?

**Comment by amaurymartiny** [comment]
As a 1st version, I'd say on error just send an email to the user with the HTTP response. I think that's what Github does tbh, go to `/settings/hooks` in any repo.

Would also be wise to see how services like Zapier handle failed webhook calls. Probably they store all webhook calls in a DB, with a dashboard to see all your past webhook deliveries & errors.

**Comment by ririko5834** [comment]
Really good idea to support webhook as receipts, there should be also support to add Discord and Slack webhooks so you can receive emails on Discord in embed format. Or option to add custom webhook URL where email subject, body, sender and other data would be POSTED in json format

**Comment by ririko5834** [comment]
It should be logged to Failed deliveries and tried to be recalled. Or there should also be an option to set up a fallback email address for webhook recipients where emails would be forwarded in case webhook doesn't return 20x status code.

**Comment by gerynugrh** [comment]
Hi @willbrowningme what do you think about this feature request? If you are okay with adding this feature I could spent some time to work on it.

**Comment by willbrowningme** [comment]
@gerynugrh It's still on the roadmap [here](https://github.com/orgs/anonaddy/projects/1/views/1?pane=issue&itemId=63479184) I just haven't had time to work on it yet.

I think using Laravel's [HTTP Client](https://laravel.com/docs/11.x/http-client#main-content) should really help with it.

There is also Spatie's webhook server package - https://github.com/spatie/laravel-webhook-server

**Comment by andre-paulo98** [comment]
Is this not considered anymore? Couldn't find it on the roadmap.
This would be huge for automation for a few services that only send email, and you want to transfer to a webhook.

Oh I just noticed its under "Rules System"

**Comment by willbrowningme** [comment]
@andre-paulo98 Yes it is still planned (likely as part of the rules system) hopefully will be able to add it in the next few months.
